src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs

```diff
@@ -44,14 +44,18 @@
    (fn [{:keys [this]}]
      (this :load-agora-method))
    :load-agora-method
-   (fn [{:keys [props state]}]
-     (endpoints/call-ajax-orch
-      {:endpoint (endpoints/get-agora-method (:namespace props) (:name props) (:snapshotId props))
-       :headers utils/content-type=json
-       :on-done (fn [{:keys [success? get-parsed-response status-text]}]
-                  (if success?
-                    (swap! state assoc :loaded-method (get-parsed-response))
-                    (swap! state assoc :error status-text)))}))})
+   (fn [{:keys [props state]} & [method-ref]]
+     (let [location (if (nil? method-ref) props method-ref)
+           namespace (:namespace location)
+           name (:name location)
+           snapshotId (:snapshotId location)]
+       (endpoints/call-ajax-orch
+         {:endpoint (endpoints/get-agora-method namespace name snapshotId)
+          :headers utils/content-type=json
+          :on-done (fn [{:keys [success? get-parsed-response status-text]}]
+                     (if success?
+                       (swap! state assoc :loaded-method (get-parsed-response))
+                       (swap! state assoc :error status-text)))})))})
 
```

Comment:
@MattPutnam  @zarsky-broad  @dmohs 
I talked to Mohs about what what was going on here and he recommended changing the method-ref argument to optional. Thoughts?